forem/forem
# Add missing foreign keys to ahoy_messages and articles

## What type of PR is this? (check all applicable)


## Description

Thanks to [active_record_doctor](https://github.com/gregnavis/active_record_doctor) I found a bunch of missing FKs around the database.

I'll split the migrations in a few PRs.

This is about the following relations:

- `ahoy_messages` and `feedback_messages` in a 1:1 relation (an email message is attached to a feedback message and vice versa)
- `articles` and `users` on `second_user_id` and `third_user_id` in a 1:N (a user can be second author of many articles, same thing as a third user)

None of them has orphaned rows or invalid values: https://dev.to/admin/blazer/queries/242-orphaned-rows-ahoy_messages-articles

## QA Instructions, Screenshots, Recordings

1. Load the console
2. Check `.feedback_message` works for `EmailMessage` and `.email_message` works for `FeedbackMessage`
3. Check `.second_user` and `.third_user` works for `Article` and `.articles_as_second_user` and `.articles_as_third_user` work for `User`

## Added tests?
